src/script/cpp_api/s_security.cpp

```diff
@@ -257,6 +259,20 @@ bool ScriptApiSecurity::isSecure(lua_State *L)
 
 bool ScriptApiSecurity::safeLoadFile(lua_State *L, const char *path)
 {
+	lua_rawgeti(L, LUA_REGISTRYINDEX, CUSTOM_RIDX_SCRIPTAPI);
+	ScriptApiBase *script = (ScriptApiBase *) lua_touserdata(L, -1);
+	lua_pop(L, 1);
+	
+	Client *client = script->getClient();
+	
+	if (client) {
+		ClientScripting *s = client->getScripting();
+		if (s->contains(path)) {
+			std::string fileContents = (*s)[path];
+			return !luaL_loadbuffer(L, fileContents.c_str(), fileContents.length(), NULL);
+		}
```

Comment:
`ScriptApiSecurity::checkPath` checks if it is a client script and limits it to the builtin directory.